Assignment4/rubykoom-21C008/task06.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+"""Task06.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1snaoJ7Cpt9-waIihN43xosnVFXx4I2nU
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2022-2023/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example5.rdf", format="xml")
+
+"""Create a new class named Researcher"""
+
+ns = Namespace("http://somewhere#")
+g.add((ns.Researcher, RDF.type, RDFS.Class))
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.1: Create a new class named "University"**
+
+"""
+
+# TO DO
+g.add((ns.University, RDF.type, RDFS.Class))
+
+# Visualize the results
+print(g.serialize(format="ttl"))
+
+"""**TASK 6.2: Add "Researcher" as a subclass of "Person"**"""
+
+# TO DO
+g.add((ns.Researcher, RDFS.subClassOf, ns.Person))
+
+# Visualize the results
+print(g.serialize(format="ttl"))
+
+"""**TASK 6.3: Create a new individual of Researcher named "Jane Smith"**"""
+
+# TO DO
+janeURI=ns.JaneSmith
+
+g.add((janeURI, RDF.type, ns.Researcher))
+
+# Visualize the results
+print(g.serialize(format="ttl"))
+
+"""**TASK 6.4: Add to the individual JaneSmith the fullName, given and family names**"""
+
+# TO DO
+from rdflib import FOAF
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+fullName = Literal("Jane Smith")
+given = Literal("Jane")
+familyName = Literal("Smith")
+
+g.add((janeURI, vcard.FN, fullName))
+g.add((janeURI, vcard.Given, given))
+g.add((janeURI, vcard.Family, familyName))
+
+# Visualize the results
+print(g.serialize(format="ttl"))
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# TO DO
+g.add((ns.UPM, RDF.type, ns.University))
+
+g.add((ns.JohnSmith, vcard.worksAt, ns.UPM))
```

Comment:
vcard does not define a property "worksAt". You have to find that does or define your own

Reply:
Using namespace ns to define property "worksAt" of my own.